iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/OperatorTreeGenerator.java

```diff
@@ -3348,4 +3358,126 @@ public Operator visitExplainAnalyze(ExplainAnalyzeNode node, LocalExecutionPlanC
+    Filter filter = context.getGlobalTimeFilter();
+    Map<IDeviceID, Boolean> deviceIDToAligned = new HashMap<>();
+    for (Map.Entry<PartialPath, Boolean> entry : node.getDevicePathsToAligned().entrySet()) {
+      deviceIDToAligned.put(new PlainDeviceID(entry.getKey().getFullPath()), entry.getValue());
+    }
+    ActiveDeviceRegionScanOperator regionScanOperator =
+        new ActiveDeviceRegionScanOperator(
+            operatorContext, node.getPlanNodeId(), deviceIDToAligned, filter);
+
+    DataDriverContext dataDriverContext = (DataDriverContext) context.getDriverContext();
+    dataDriverContext.addSourceOperator(regionScanOperator);
+    dataDriverContext.setDeviceIDToAligned(deviceIDToAligned);
+    dataDriverContext.setQueryDataSourceType(QueryDataSourceType.DEVICE_REGION_SCAN);
+
+    return regionScanOperator;
+  }
+
+  @Override
+  public Operator visitTimeSeriesRegionScan(
+      TimeseriesRegionScanNode node, LocalExecutionPlanContext context) {
+    OperatorContext operatorContext =
+        context
+            .getDriverContext()
+            .addOperatorContext(
+                context.getNextOperatorId(),
+                node.getPlanNodeId(),
+                ActiveTimeSeriesRegionScanOperator.class.getSimpleName());
+    Filter filter = context.getGlobalTimeFilter();
+    DataDriverContext dataDriverContext = (DataDriverContext) context.getDriverContext();
+
+    Map<IDeviceID, Map<String, TimeseriesSchemaInfo>> timeseriesToSchemaInfo = new HashMap<>();
+    for (Map.Entry<PartialPath, Map<PartialPath, List<TimeseriesSchemaInfo>>> entryMap :
+        node.getDeviceToTimeseriesSchemaInfo().entrySet()) {
+      Map<String, TimeseriesSchemaInfo> timeseriesSchemaInfoMap =
+          getTimeseriesSchemaInfoMap(entryMap, dataDriverContext);
+      timeseriesToSchemaInfo.put(
+          new PlainDeviceID(entryMap.getKey().getFullPath()), timeseriesSchemaInfoMap);
+    }
+    ActiveTimeSeriesRegionScanOperator regionScanOperator =
+        new ActiveTimeSeriesRegionScanOperator(
+            operatorContext, node.getPlanNodeId(), timeseriesToSchemaInfo, filter);
+
+    dataDriverContext.addSourceOperator(regionScanOperator);
+    dataDriverContext.setQueryDataSourceType(QueryDataSourceType.TIME_SERIES_REGION_SCAN);
+
+    return regionScanOperator;
+  }
+
+  private static Map<String, TimeseriesSchemaInfo> getTimeseriesSchemaInfoMap(
+      Map.Entry<PartialPath, Map<PartialPath, List<TimeseriesSchemaInfo>>> entryMap,
+      DataDriverContext context) {
+    Map<String, TimeseriesSchemaInfo> timeseriesSchemaInfoMap = new HashMap<>();
+    for (Map.Entry<PartialPath, List<TimeseriesSchemaInfo>> entry :
+        entryMap.getValue().entrySet()) {
+      PartialPath path = entry.getKey();
+      context.addPath(path);
+      if (path instanceof MeasurementPath) {
+        TimeseriesSchemaInfo timeseriesSchemaInfo = entry.getValue().get(0);
+        timeseriesSchemaInfo.setDataBase(path.getDevicePath().getDevice());
```

Comment:
database is not the previous level of device Path, what's the database name of `root.db.a..b.c.d1.s1`?

Reply:
one region belongs to one database, maybe you can get that from DataRegion.